dashboard/app/models/script.rb

```diff
@@ -214,25 +214,40 @@ def self.stage_extras_script_ids
   # @param [User] user
   # @return [Script[]]
   def self.valid_scripts(user)
-    user_experiments_enabled = Course.has_any_course_experiments?(user)
-    with_hidden = !user_experiments_enabled && user.hidden_script_access?
-    cache_key = "valid_scripts/#{with_hidden ? 'all' : 'valid'}"
-    scripts = Rails.cache.fetch(cache_key) do
-      Script.
-          all.
-          select {|script| with_hidden || !script.hidden}
-    end
+    has_any_course_experiments = Course.has_any_course_experiments?(user)
+    with_hidden = !has_any_course_experiments && user.hidden_script_access?
+    scripts = with_hidden ? all_scripts : visible_scripts
 
-    if user_experiments_enabled
+    if has_any_course_experiments
       scripts = scripts.map do |script|
         alternate_script = script.alternate_script(user)
         alternate_script.presence || script
       end
     end
 
+    if !with_hidden && has_any_pilot_access?(user)
+      scripts = scripts.concat(all_scripts.select {|s| s.has_pilot_access?(user)})
+    end
+
     scripts
   end
 
+  class << self
+    private
```

Comment:
this is just a way to define class methods. I'm doing it this way instead of e.g. `def self.all_scripts` because this lets me mark them private. upon further research, it is also possible to accomplish this via `private_class_method def self.all_scripts` and omit the `class << self`. Let me know if you prefer that.

Reply:
TIL! I've always used `private_class_method` but totally cool with this if it's a Ruby idiom.